### PR TITLE
RHCLOUD-35528 Consolidate resources and relationships into Resources and Relationships tables and preparing for future work

### PR DESCRIPTION
**Pull request by josejulio**
- Use the same table for all resources This would help with custom resources
 - Keep multiple proto-services for now They help with the data integrity of resource_data content
 - The same logic applies for everything If we decide that specific logic should be applied for different resources we could use the old pattern and only create specifics for said components
- Adding resourcesHistory and RelationshipHistory for future work
- Creating specific eventing types instead of relying on maps

Note that this does not yet make use of the history tables - PR is already big enough.
The focus here was to define the new schema/model and wire everything that was touched in the process

### Database schema

kessel-inventory-db drawio

#### Resource

```json
{
   "specversion":"1.0",
   "type":"redhat.inventory.resources.rhel-host.created",
   "source":"",
   "id":"be148d91-871c-11ef-b169-a63ce1be7759",
   "subject":"/resources/rhel-host/1",
   "time":"2024-10-10T09:31:38.951828234-06:00",
   "datacontenttype":"application/json",
   "data":{
      "metadata":{
         "id":1,
         "resource_type":"rhel-host",
         "last_reported":"2024-10-10T09:31:38.951828234-06:00",
         "workspace":"workspace1"
      },
      "reporter_data":{
         "reporter_instance_id":"curl/8.6.0",
         "last_reported":"2024-10-10T09:31:38.951828234-06:00",
         "reporter_type":"OCM",
         "console_href":"www.example.com",
         "api_href":"www.example.com",
         "local_resource_id":"1",
         "reporter_version":"0.1"
      }
   }
}
```

#### Relationship

```json
{
   "specversion":"1.0",
   "type":"redhat.inventory.resources-relationship.k8s-policy_is-propagated-to_k8s-cluster.created",
   "source":"",
   "id":"cccf7848-871c-11ef-b169-a63ce1be7759",
   "subject":"/resources-relationship/k8s-policy_is-propagated-to_k8s-cluster/1",
   "time":"2024-10-10T09:32:03.664262907-06:00",
   "datacontenttype":"application/json",
   "data":{
      "metadata":{
         "id":1,
         "relationship_type":"k8s-policy_is-propagated-to_k8s-cluster",
         "last_reported":"2024-10-10T09:32:03.664262907-06:00"
      },
      "reporter_data":{
         "reporter_type":"OCM",
         "subject_local_resource_id":"1",
         "object_local_resource_id":"2",
         "reporter_version":"0.1",
         "reporter_instance_id":"curl/8.6.0"
      },
      "resource_data":{
         "status":3
      }
   }
}
```

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: internal/eventing/api/event.go
Comment (randymgeorge) [inline internal/eventing/api/event.go:1]: For the ResourceMetadata, do you not want this to be workspaceID vs just workspace? line 40 and line 118

File: internal/service/common/common.go
Comment (randymgeorge) [inline internal/service/common/common.go:1]: is line 16 still metadata.workspace for this has not been updated in the api/protobuf definition yet?

**Review by randymgeorge — COMMENTED** [review]
see specific file comments

File: internal/service/common/common.go
Comment (josejulio) [reply internal/service/common/common.go:1]: That's right - i will need to dive there to update those resources and align them with the latest info from the docs [PR](https://github.com/project-kessel/docs/pull/6/files).

File: internal/eventing/api/event.go
Comment (josejulio) [reply internal/eventing/api/event.go:1]: yes, haven't made any change related to  https://github.com/project-kessel/docs/pull/6/files

It makes sense to align with the same terminology as much as possible.

**Review by randymgeorge — APPROVED** [review]

**Comment by akoserwal** [comment]
Looks good, few questions

* How will the inventory API get the `org_id`?
* How will the inventory API RelationshipData/RelationshipType?
* Why SubjectId  and ObjectId are same?

**Comment by josejulio** [comment]
> * How will the inventory API get the `org_id`?

We will update the API to receive that information

> * How will the inventory API will get the RelationshipData/RelationshipType?
 
Right now that will be api dependant i.e. rhel host doesn't have anything right now but k8s-cluster has the data

> * Why SubjectId  and ObjectId are same?

wdym ? Each of those are the resource.id of each side of the relationship

**Comment by akoserwal** [comment]
> > * How will the inventory API get the `org_id`?
> 
> We will update the API to receive that information
> 
> > * How will the inventory API will get the RelationshipData/RelationshipType?
> 
> Right now that will be api dependant i.e. rhel host doesn't have anything right now but k8s-cluster has the data
> 
> > * Why SubjectId  and ObjectId are same?
> 
> wdym ? Each of those are the resource.id of each side of the relationship

 SubjectId  and ObjectId point to same `resource.id` why we are mapping to same resource.id?

**Comment by josejulio** [comment]
> SubjectId and ObjectId point to same `resource.id` why we are mapping to same resource.id?

ah - the resource resolution (i.e. from local_resource_id to our resource_id) is not in our codebase right now - we are using "0" as the resource_id at this point but a followup pr should change it.

**Pull request merged**